### PR TITLE
Remove ChoiceField in favor of using Checkbox/Radio directly

### PR DESCRIPTION
**Pull request by wishdasher**
After talking with Kevin, we decided to put the label and description props right on Checkbox and Radio. To avoid redundant implementation, Checkbox and Radio now both wrap ChoiceInternal. Choice is still a wrapper meant for use in CheckboxGroup and RadioGroup.

I've updated the docs and examples to reflect this change in expected usage.

I've also improved some minor stuff: removing the redundant `style` prop from the core components, removing an unnecessary `View` wrapper, and stop passing `value` around where it's unnecessary.

### CODE REVIEW

**Comment by jeresig** [comment]
Deploy preview for *wonder-blocks* ready!

Built with commit 505ee89673707e875fc401c834605e2630946019

https://deploy-preview-248--wonder-blocks.netlify.com

File: packages/wonder-blocks-form/components/choice-internal.js [hunk 1]
```diff
@@ -31,36 +22,42 @@ type Props = {|
...
- * props auto-populated, but ChoiceField does not.
- */
-export default class ChoiceInternal extends React.Component<Props> {
+ * This is a potentially labeled 🔘 or ☑️ item. This is an internal component
+ * that's wrapped by Checkbox and Radio. Choice is a wrapper for Checkbox and
+ * Radio with many of its props auto-populated, to be used with CheckboxGroup
+ * and RadioGroup. This design allows for more explicit prop typing. For
+ * example, we can make onChange a required prop on Checkbox but not on Choice
+ * (because for Choice, that prop would be auto-populated by CheckboxGroup).
+ */ export default class ChoiceInternal extends React.Component<Props> {
```
Comment (jeresig) [inline packages/wonder-blocks-form/components/choice-internal.js]: 😢 silly prettier

File: packages/wonder-blocks-form/components/checkbox-core.js [hunk 2]
```diff
@@ -63,11 +59,10 @@ export default class CheckboxCore extends React.Component<Props> {
 
         const props = {
             "data-test-id": testId,
-            ...handlers,
         };
 
         return (
-            <View style={[sharedStyles.wrapper, style]}>
+            <React.Fragment>
```
[the remark this answers is not in the document]
Comment (wishdasher) [reply packages/wonder-blocks-form/components/checkbox-core.js]: Yes, it's being applied on the ChoiceInternal level. So the user can add extra padding/margin/styles to the entire component.

[the remark this answers is not in the document]
Comment (wishdasher) [reply packages/wonder-blocks-form/components/checkbox-core.js]: I was previously erroneously applying it twice.

[the remark this answers is not in the document]
Comment (jeresig) [reply packages/wonder-blocks-form/components/checkbox-core.js]: Ohh gotcha - great!

**Pull request merged**